config/routes.rb

```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
+  devise_for :users
   get 'main_controller/index'
-
+  root 'main_controller#index'
```

Comment:
Change with `root 'main#index'` there is no need to specify that main is a controller